schedule/views.py

```diff
@@ -8,25 +7,41 @@
 from forecaster.forecaster import forecast as c2alg2 # company 2 alg 2
 
 from c1algo1 import scheduler as c1alg1# company 1 alg 1
-from c1algo2.forecaster import forecast as c1algo2
+from c1algo2.forecaster import forecast as c1alg2
 
+from schedule.alg_data_generator import get_historic_course_data
+from schedule.alg_data_generator import get_program_enrollment_data
+from schedule.alg_data_generator import get_schedule
+from schedule.alg_data_generator import get_professor_dict_mock
+from schedule.alg_data_generator import get_schedule_alg1_mock
+from schedule.alg_data_generator import get_schedule_alg2_mock
+
+import traceback
+import json
+
+from courses.models import Course
 
 
 class Schedule(APIView):
     # GET / schedule / {year - semester}
     def get(self, request: HttpRequest, year: int, semester: str, requested_company_alg: int) -> HttpResponse:
-        alg2_result = c2alg1(None, None, None) if requested_company_alg == 1 else c2alg2(None, None, None)
-        alg1_result = c1alg1.generate_schedule() if requested_company_alg == 1 else c2alg1(None, None, None)
-
-        # these 2 lines are temporary and just to make unit tests pass while integration is in progress
-        alg2_result = "OK"
-        alg1_result = "OK"
-
-        if "OK" == alg1_result and "OK" == alg2_result:
-            body = "GENERATED SCHEDULE FROM COMPANY %d ALGORITHM, AND BOTH RETURNED OK" % requested_company_alg
-            return HttpResponse(body, status=status.HTTP_200_OK)
-        else: 
-            return HttpResponse(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        # Create params for algorithms packages
+        historical_data = get_historic_course_data()
+        previous_enrollment = get_program_enrollment_data()
+        schedule = get_schedule_alg2_mock()
+        professors = get_professor_dict_mock()
+        schedule_1 = get_schedule_alg1_mock()
```

Comment:
Can we use the output from alg 2? It's better for the demo if we can say the process is similar to what will happen in the end product.

Reply:
We totally can when it's just company 2. I was just using the alg1 mock because company 1's
 algo2 just returns "OK"

Reply:
Can you please update it for our company? IMO you should do the same for their company and just let the call fail if their algorithm isn't working. I don't think you should implement a workaround because it hides the problem.